src/pages/index.js

```diff
@@ -39,6 +39,18 @@ const popupContentGatsby = `
 
 const IndexPage = () => {
   const markerRef = useRef();
+  const [location,setLocation] = useState(LOCATION);
```

Comment:
so i think i'd like to avoid using state for this so we can avoid triggering a rerender of the map component. from my experience with doing that in the past, it can mess up the state of the map and can cause bugs managing it. i think what we can do instead is if you move the `getCurrentPosition` handler inside of `mapEffect`, you should be able to trigger a recenter of the map using:
```
leafletElement.flyTo(center)
```
https://leafletjs.com/reference-1.5.0.html#map-flyto